resolver/changer_test.go

```diff
@@ -0,0 +1,99 @@
+package resolver
+
+import (
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+)
+
+func TestMasterChanger(t *testing.T) {
```

Comment:
I'd prefer that we test `OnMasterChanged` and `UpdatedMasters` separately. This test is confusing.